src/command/init/mod.rs

```diff
@@ -13,12 +22,84 @@ impl Init {
     pub async fn run(&self) -> RoverResult<RoverOutput> {
         println!("\nWelcome! This command helps you initialize a new GraphQL API project using Apollo Federation with Apollo Router.\n");
 
+        let request_service = ReqwestService::builder().build()?;
+
         let use_case = self.use_case_options.get_or_prompt_use_case()?;
         match use_case {
-            ProjectUseCase::Connectors => println!("\nComing soon!\n"),
+            ProjectUseCase::Connectors => {
+                let repo_url = "https://github.com/apollographql/rover-connectors-starter/archive/refs/heads/main.tar.gz";
+                self.init_project(repo_url, request_service).await?;
+            }
             ProjectUseCase::GraphQLTemplate => println!("\nComing soon!\n"),
         }
 
         Ok(RoverOutput::EmptySuccess)
     }
+
+    fn prompt_creation(&self, artifacts: Vec<String>) -> std::io::Result<bool> {
+        infoln!("The following files will be created:");
+        let mut top_level_artifacts = HashSet::new();
+        artifacts
+            .iter()
+            .filter(|path| path.matches('/').count() == 1 || path.matches('/').count() == 0)
+            .sorted()
+            .for_each(|path| {
+                top_level_artifacts.insert(path.clone());
+            });
+
+        for artifact in top_level_artifacts {
+            infoln!("{}", artifact);
+        }
+        println!();
+
+        prompt_confirm_default_yes("Proceed with creation?")
+    }
+
+    async fn init_project(&self, repo_url: &str, http_service: ReqwestService) -> RoverResult<()> {
+        let template = TemplateFetcher::new(http_service)
+            .call(repo_url.parse()?)
+            .await?;
+        let current_dir = env::current_dir()?;
+        let current_dir = Utf8PathBuf::from_path_buf(current_dir)
+            .map_err(|_| anyhow::anyhow!("Failed to parse current directory"))?;
+
+        let output_path = match env::var("INIT_OUTPUT_DIR") {
+            Ok(value) => Utf8PathBuf::from(value),
+            Err(_) => current_dir,
+        };
+
+        //TODO: move this in favor of the template command handling
+        match read_dir(&output_path) {
+            Ok(mut dir) => {
+                if dir.next().is_some() {
+                    let mut err = RoverError::new(anyhow!(
+                        "Cannot initialize the project because the '{}' directory is not empty.",
+                        &output_path
+                    ));
+                    err.set_suggestion(RoverErrorSuggestion::Adhoc(
+                        "Please run Init on an empty directory".to_string(),
+                    ));
+                    return Err(err);
```

Comment:
Nitpick - I prefer the chained vesion rather than using a temporary variable:

```suggestion
                    return Err(RoverError::new(anyhow!(
                        "Cannot initialize the project because the '{}' directory is not empty.",
                        &output_path
                    )).with_suggestion(RoverErrorSuggestion::Adhoc(
                        "Please run Init on an empty directory".to_string(),
                    )));
```